### ISSUE TITLE
format docx affects pdf output

### ISSUE THREAD

**Issue by MarcoPortmann** [quarto-dev/quarto-cli#9040]
### Bug description

I have a modelsummary table with panels. However, when rendering the pdf, I notice that the panel headers are not displayed if the the yaml header contains the format docx in addition to format pdf
```markdown
docx:
    documentclass: scrreport
````
 If I delete the two lines, the table including panel headings appears in the pdf.

### Steps to reproduce

``````markdown
---
format:
  docx:
    documentclass: scrreport
  html:
    toc: true
    toc-depth: 3
    toc-location: right
    number-sections: true
  pdf:
    documentclass: scrreport
    toc: true
    toc-depth: 3
    number-sections: true
    fig-cap-location: top
---

```{r message=TRUE, include=FALSE}
library(tinytable)
library(modelsummary)
```




```{r}
#| echo: FALSE
#| label: tbl-test
#| tbl-cap: "Test table"

panels <- list(
    "Panel A: MPG" = list(
        "A" = lm(mpg ~ hp, data = mtcars),
        "B" = lm(mpg ~ hp + factor(gear), data = mtcars)),
    "Panel B: Displacement" = list(
        "A" = lm(disp ~ hp, data = mtcars),
        "C" = lm(disp ~ hp + factor(gear), data = mtcars))
)

modelsummary(
    panels,
    shape = "rbind",
    gof_map = c("nobs", "r.squared"))

```

``````

### Expected behavior

With this yaml header
````makdown
---
format:
  html:
    toc: true
    toc-depth: 3
    toc-location: right
    number-sections: true
  pdf:
    documentclass: scrreport
    toc: true
    toc-depth: 3
    number-sections: true
    fig-cap-location: top
---
````

the pdf output looks like this:
![image](https://github.com/quarto-dev/quarto-cli/assets/7468121/461cb689-78e2-4f6e-944f-ac248a41d956)

### Actual behavior

However, the pdf output looks like this if docx is present in the yaml header:
![image](https://github.com/quarto-dev/quarto-cli/assets/7468121/0d661a3a-5229-461d-9e8d-482f4bd64921)

### Your environment

RStudio 2023.12.0 Build 369

### Quarto check output

Quarto 1.4.551
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.4.551
      Path: /opt/quarto/bin

[✓] Checking tools....................OK
      TinyTeX: (not installed)
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /usr/local/bin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /tmp/Rtmp4mNj1j/rstudio/terminal/python3
      Jupyter: (None)

      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter

[✓] Checking R installation...........OK
      Version: 4.3.3
      Path: /usr/local/lib/R
      LibPaths:
        - /usr/local/lib/R/site-library
        - /usr/local/lib/R/library
        - /home/pco2/R/x86_64-pc-linux-gnu-library/4.3
      knitr: 1.45
      rmarkdown: 2.26

[✓] Checking Knitr engine render......OK

**Comment by cscheid** [comment]
```
  docx:
    documentclass: scrreport
```

I'm not sure what you're expecting to get with this option since `documentclass` a latex-only option. I imagine this is confusing some downstream processing. Why did you add it?

**Comment by MarcoPortmann** [comment]
You are right, the `documentclass: scrreport` was confusing and unnecessary. I have revised the code. However, the same problem also occurs with `docx: default` which makes more sense, I hope. I must admit, I have not worked much with docx so far.

**Comment by cscheid** [comment]
Hm - knitr is definitely emitting completely different output here. I don't know why that's the case. @cderv?
